### PR TITLE
[Chore] Update some dependencies, cleanup

### PR DESCRIPTION
**Pull request by gromakovsky**
## Description

Routine changes to support newer GHC versions.

## Related issues(s)

None

## ✓ Checklist for your Pull Request

Ideally a PR has all of the checkmarks set.

If something in this list is irrelevant to your PR, you should still set this
checkmark indicating that you are sure it is dealt with (be that by irrelevance).

- [x] I made sure my PR addresses a single concern, or multiple concerns which
      are inextricably linked. Otherwise I should open multiple PR's.
- [x] If I added/removed/deprecated functions/re-exports,
      I checked whether these changes impact the [`.hlint.yaml`](https://github.com/serokell/universum/tree/master/.hlint.yaml) rules
      and updated them if needed.

#### Related changes (conditional)

- Tests

  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation

  I checked whether I should update the docs and did so if necessary:

  - [x] [README](https://github.com/serokell/universum/tree/master/README.md)
  - [x] Haddock

- Record your changes

  - [x] I added an entry to the [changelog](https://github.com/serokell/universum/tree/master/CHANGES.md) (creating the `Unreleased` section if necessary) if my changes are visible to the users
        and
  - [x] provided a migration guide for breaking changes if possible

#### Stylistic guide (mandatory)

- [x] My commit history is clean (only contains changes relating to my
      issue/pull request and no reverted-my-earlier-commit changes) and commit
      messages start with identifiers of related issues in square brackets.

  **Example:** `[#42] Short commit description`

  If necessary both of these can be achieved even after the commits have been
  made/pushed using [rebase and squash](https://git-scm.com/docs/git-rebase).

## ✓ Release Checklist

- [x] I updated the version number in `universum.cabal`.
- [x] I updated the [changelog](https://github.com/serokell/universum/tree/master/CHANGES.md) and moved everything
      under the "Unreleased" section to a new section for this release version.
- [x] If any definitions (functions, type classes, instances, etc) were added,
      I added [`@since` haddock annotations](https://haskell-haddock.readthedocs.io/en/latest/markup.html#since).
- [ ] (After merging) I created a new entry in the [releases](https://github.com/serokell/universum/releases) page,
      with a summary of all user-facing changes.
    *  I made sure a tag was created using the format `vX.Y.Z`

### CODE REVIEW

File: test/Test/Universum/StringProps.hs [hunk 1]
```diff
@@ -1,3 +1,8 @@
+-- Some tests in this module have a caveat: they depend on compiler
+-- optimizations and will likely fail if compiled with -O0.
+-- That's because they test are rewrite rules, and these rules
```
Comment (Martoon-00) [inline test/Test/Universum/StringProps.hs:3]: Something seems to be amiss in this sentence

File: universum.cabal [hunk 2]
```diff
@@ -107,7 +107,7 @@ library
                      , stm
                      -- Make sure that "toString-toText-rewritting" note
                      -- is still valid when bumping this constraint.
-                     , text >= 1.0.0.0 && <= 2.0.2
+                     , text >= 1.0.0.0 && <= 2.1.1
```
Comment (Martoon-00) [inline universum.cabal:110]: I checked this change:
1. There seems to be no changes to `pack`/`unpack` functions or their rewrite rules in `text` package. Please update the commit mentioned in `toString-toText-rewritting` note in Haskell code to `c826a6cb3d29242ce3632b490670ae7997689eaa` (current `origin/master` of text package)
2. Benchmarks produce the expected result.

File: test/Test/Universum/StringProps.hs (hunk 1 above)
Comment (gromakovsky) [reply test/Test/Universum/StringProps.hs]: Looks weird, I don't remember what exactly I wanted to write. Anyway, pushed a fix with sane wording.

File: universum.cabal (hunk 2 above)
Comment (gromakovsky) [reply universum.cabal:110]: Added "in place" (in the old commit).

File: test/Test/Universum/StringProps.hs [hunk 3]
```diff
@@ -1,3 +1,8 @@
+-- Some tests in this module have a caveat: they depend on compiler
+-- optimizations and will likely fail if compiled with -O0.
+-- That's because they depend on rewrite rules (and tests these rules),
```
Comment (Martoon-00) [inline test/Test/Universum/StringProps.hs:3]:
```suggestion
-- That's because they depend on rewrite rules (and test these rules),
```

(I guess)

**Review by Martoon-00 — APPROVED** [review]
One more typo as it seems to be, other than that LGTM, approving.

File: test/Test/Universum/StringProps.hs (hunk 3 above)
Comment (gromakovsky) [reply test/Test/Universum/StringProps.hs]: Right, done, going to merge and make a release once CI passes.

**Pull request merged**